### ISSUE TITLE
Add support for Tenderly through automation

### ISSUE THREAD

**Issue by jbunce** [omisego/plasma-contracts#474]
```
[ ] bug report
[x] feature request
```

## Current Behavior
We have to push deployed contracts to Tenderly manually for monitoring. This means if it's forgotten and we lose the build artifacts we don't have contract level monitoring for that environment.

## Suggested Fix
This is intended to be achieved through a module/migration within Truffle.

1) Introduce a flag that disables Tenderly (assume that a Tenderly deployment is desired by default). Suggestion: `export TENDERLY=false`
2) Add support for obtaining the Tenderly authentication token. Suggestion: `export TENDERLY_TOKEN=foo`
3) Add support for creating the Tenderly project: `export TENDERLY_PROJECT=bar`
4) Add support for creating the Tenderly project:
```
tenderly init --create-project --project $TENDERLY_PROJECT
```
5) After the contracts have been deployed issue:
```
tenderly login --authentication-method=token --token=$TENDERLY_TOKEN
```
6) Push the contracts to Tenderly:
```
tenderly push
```

This would fit with the work in https://github.com/omisego/devops/issues/242

**Comment by thec00n** [comment]
@jbunce It's not clear to be how the tenderly integration steps fit in the current repository and how this is then executed during deployment. Is it that we define the steps in package.json and then you call them sequentially?

```
1.) install tenderly
2.) set env variables
3.) create the project 
4.) truffle migrate
5.) tenderly login 
6.) tenderly push 
```

**Comment by jbunce** [comment]
Hey @thec00n. Without actually going through the process myself while implementing this I'm not sure. If you want to take a stab at this let me know and I'll see if I can help.

**Comment by jbunce** [comment]
From a conversation @thec00n and I had in person:

- We've agreed to facilitate this for development environments
- Manual process for production environments
- Implementation in JS with exec/subprocess
- `plasma-contracts` change required so we don't pass authority key material into Truffle

Feel free to add if I've missed anything :)

**Comment by thec00n** [comment]
Closing this issue as it addressed with #493. Splitting up dev/prod deployments is tracked here https://github.com/omisego/plasma-contracts/issues/499
